src/yadism/coefficient_functions/fonll/g1_nc.py

```diff
@@ -0,0 +1,82 @@
+from ..special.zeta import zeta2, zeta3
+from ..special.nielsen import nielsen
+from eko.constants import CA as ca
+from eko.constants import CF as cf
+from eko.constants import TR as tr
+
+# complex :math:`S_{n,m}(x)`-> nielsen(N, M, X)
+# for the light coefficient function; singlet, non singlet, gluonic 
+#LS= LNS + LPS
+# for the heavy coefficient function; singlet (only pure singlet contributes), gluonic 
+# HS= HPS
+
+@nb.njit("f8(f8,f8[:])", cache=True)
+class NonSinglet():
+    def NNLO(z, args):
+        """Massive asymptotic polarized flavor non–singlet Wilson coefficient in the MSbar scheme. 
+
+        |ref| implements :eqref:`254`, :cite:`Blümlein` (needs to be added). (coupling to the light quark)
+        """
+        L = args[0]
+        dlz=np.log(z) #H(0;x)
+        dlz2 =dlz*dlz  #H(0;x)^2
+        dlm = (-1)* np.log(1.0e0-z) #H(1;x) (relative to the definitions in other scripts, theres an extra (-1), im defining it according to 'Harmonic Polylogarithms' E. Remiddia,b and J. A. M. Vermaseren )
+        dlm2 =dlm*dlm
+
+        res= cf*tr*((-4/3)* np.power(L,2)*(1+ z) + (4/27)* (-47-218*z+36* zeta2 +36*z*zeta2)+ ((265/9)+((2*L)/3) + 2*np.power(L,2))*(np.delta(1-z)) - 4*(1 + z)*dlz2 +L *((-8/9)*(-1 + 11*z) - (8/3)*(1 + z) *dlz) - (8/9)*(5 + 14*z) * dlm - (4/3)* (1 + z)* dlm2+(-(8/9)*(13 + 28*z)-(8/3)*(1 + z)*dlm) - (8/3)*(1 + z)* nielsen(0,1,z) + 1/(1-z)*((8*np.power(L,2))/3 - (2/27)*(-359 + 144*zeta2) + 8*dlz + L*((80/9)+(16/3)*dlz) + (116/9)*dlm + (8/3)*dlm2 + dlz*((268/9) + (16/3)*dlm) + (16/3)* nielsen(0,1,z))) 
+        #remove delta later
+        return res
+    
+
+    class PureSinglet():
+        def NNLO(z, args):
+            """Massive asymptotic polarized pure singlet Wilson coefficient with in the Larin scheme. 
+
+            |ref| implements :eqref:`261`, :cite:`Blümlein` (needs to be added). (coupling to the heavy quark)
+            """
+            L = args[0]
+            dlz=np.log(z) #H(0;x)
+
+            dlm = (-1)* np.log(1.0e0-z) #H(1;x) (relative to the definitions in other scripts, theres an extra (-1), im defining it according to 'Harmonic Polylogarithms' E. Remiddia,b and J. A. M. Vermaseren )
+            dlp=np.log(1.0e0+z) #H(-1;x)
+            nH001z= None#reexpress in terms of lower weight harmonic polylogarithm
+            nH011z= None #reexpress in terms of lower weight harmonic polylogarithm
+
+            res= cf*tr*(-(592/3)*(-1 + z)-(32/3)*(9 - 3*z + np.power(z,2))*zeta2 + 16*(1 + z)* zeta3 + (-1*(256/3)*(-2 + z) - 32*(1 + z)*zeta2)* dlz - (32*np.power(1+z,3)*dlp*dlz)/(3*z) + (8/3)*(21+2*np.power(z,2))*np.power(dlz,2)+(16/3)*(1+ z)*np.power(dlz,3) + np.power(L,2)*(20*(-1 + z)-8*(1 + z)*dlz) + L*((-8)*(-1 + z)+8*(-1+3*z)*dlz- 8*(1 + z)*np.power(dlz,2))+(-88*(-1+z)-80*(-1+z)*dlz)*dlm-20*(-1+z)*np.power(dlm,2)+(32*np.power(1+z,3)*nielsen(0,-1,z))/(3*z)+(16*(-1+3*z)+ 32*(1+z)*dlz)*nielsen(0,1,z)-32*(1 + z)*nH001z+16*(1 + z)*nH011z) 
+            return res
+
+    class Gluonic_light():
```

Comment:
I'm not 100% sure this piece would be needed. 
In our definition heavy is something  fully proportional to the charge of the heavy quark,
while this would only contribute to light...